### DISCUSSION TITLE
Query: Custom Icons for Custom Activities

### DISCUSSION THREAD

**Discussion by kiebor81** [elsa-workflows/elsa-core#6033] [Q&A] [answered]
I have 50+ custom activities I built for Elsa 2. These are all used in programmatic workflows because I didn't like the old implementation of the studio. The Elsa 3 studio is much nicer and I'm in the process of migrating across. Is there anyway in code, by way of embedded resources, an attribute for dataurl or byte array, etc, that allows me to use custom icons for these in the studio, or is the approach similar to the previous version where I'd need to put together some kind of front end manifest and inject them? Sorry if this is answered somewhere else. I couldn't find anything in the documentation.

**Comment by Nekrosis1** [comment, marked as answer]
There is an interface for adding custom colors and icons to activities
IActivityDisplaySettingsProvider
you can create a custom provider inheriting from that and registering it with services.AddActivityDisplaySettingsProvider<YourCustomProvider>();
You should be able to add your custom resources/icons to your custom activities with that.

**Reply by kiebor81** [reply]
Thanks.

When implementing the interface

```c#   
public class CustomDisplayProvider : IActivityDisplaySettingsProvider
    {
        public IDictionary<string, ActivityDisplaySettings> GetSettings()
        {
            //
        }
    }
```

Assume the dictionary key (string) is the activity name? And the value is

```c#
    public ActivityDisplaySettings(string color, string? icon = default)
    {
        Color = color;
        Icon = icon;
    }
```

Where icon is a path to the icon (or something else?) and color is hex code?

**Reply by valsov** [reply]
Hi, have a look at this file for examples on how to use this interface: [Elsa Studio](https://github.com/elsa-workflows/elsa-studio/blob/96a5e4ee6e9e2431fa71e3ecbe89d0fe826ac2a7/src/modules/Elsa.Studio.Workflows.Core/UI/Providers/DefaultActivityDisplaySettingsProvider.cs).
I implemented this and it worked without issue.

**Reply by kiebor81** [reply]
@valsov, thank you very much. Appreciate the response.
